source/lib/src/gpu/tabulate.cu

```diff
@@ -110,8 +139,15 @@ __forceinline__ __device__ FPTYPE dot(FPTYPE ll[4], FPTYPE rr[4]) {
 
 template <typename FPTYPE>
 __forceinline__ __device__ void warp_reduce(FPTYPE& val) {
+#if GOOGLE_CUDA
   for (int offset = 16; offset > 0; offset >>= 1) {
     val += __shfl_down_sync(FULL_MASK, val, offset);
+#elif TENSORFLOW_USE_ROCM
+  for (int offset = 32; offset > 0; offset >>= 1) {
+    val += __shfl_down(val, offset);  // ########????
+#else
```

Comment:
Should the initial `offset` be `WARP_SIZE / 2`? (since it's not fixed in rocm)

Reply:
Yes, It should be WARP_SIZE / 2 according to the metioned Torch PR